src/components/learning-units-list/LearningUnitListItem.js

```diff
@@ -1,21 +1,42 @@
 import useCompletedLearningUnit from '@hooks/useCompletedLearningUnit';
 import React, { useState, useEffect } from 'react';
 import Image from 'next/image';
+import Link from 'next/link';
 import { Button } from 'primereact/button';
 import { ToggleButton } from 'primereact/togglebutton';
 import { Card } from 'primereact/card';
 import styles from '@styles/LearningUnitsList.module.scss';
 import profilePic from '@utils/images/unit.jpeg';
 
 function LearningUnitItem({ unit }) {
-  const { data, isLoading, isError } = useCompletedLearningUnit(unit.id);
-
   const [isCompleted, setCompleted] = useState(false);
 
+  const { data, isLoading, isError } = useCompletedLearningUnit(unit.id);
+
   useEffect(() => {
     data ? setCompleted(data.completed) : false;
   }, [data]);
 
+  const handleOnChange = (clicked) => {
+    if (clicked.value) {
+      const requestOptions = {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+      };
+      fetch(`http://localhost:3001/api/learning_units/${unit.id}/completed`, requestOptions)
+        .then((response) => response.json())
+        .then(() => setCompleted(clicked.value));
```

Comment:
then() must use the response as input to change the state

Reply:
Done